src/frontend/src/components/DynamicTooltip.tsx

```diff
@@ -0,0 +1,40 @@
+import { Tooltip, Typography } from '@mui/material';
+
+interface DynamicTooltipProps {
+  title: string;
+  children: React.ReactElement;
+}
+
+const DynamicTooltip: React.FC<DynamicTooltipProps> = ({ title, children }) => {
+  return (
+    <Tooltip
+      id="tooltip"
+      placement="right"
+      title={<Typography>{title}</Typography>}
+      PopperProps={{
+        popperOptions: {
+          modifiers: [
+            {
+              name: 'flip',
+              options: {
+                fallbackPlacements: ['top', 'bottom']
+              }
+            },
+            {
+              name: 'offset',
+              options: {
+                // I think this should allow a plain value as well, not functions
+                offset: [0, -10] // distance: 10px
```

Comment:
It still can get jittery, idk :( try searching "flicker" or "jitter" or something in the [popper issues](https://github.com/floating-ui/floating-ui/issues) and see if anything works. If nothing works then it's not too big of a problem tbh

Reply:
Seems like at the exact boundary where it switches from right to top is where it can't decide and flickers. I'm guessing the offset just moved where that happens by 10px. I'll look into it some more and see if I can find a solution.